tchannel-core/src/main/java/com/uber/tchannel/schemes/JSONSerializer.java

```diff
@@ -47,14 +50,15 @@ public String decodeEndpoint(ByteBuf arg1) {
     public Map<String, String> decodeHeaders(ByteBuf arg2) {
         String headerJSON = arg2.toString(CharsetUtil.UTF_8);
         arg2.release();
-        return new Gson().fromJson(headerJSON, HEADER_TYPE);
+        Map<String, String> headers = GSON.fromJson(headerJSON, HEADER_TYPE);
+        return (headers != null) ? headers : new HashMap<String, String>();
```

Comment:
Also what happens on the encoding side during null, should we use EmptyByteBuf instead of nulls since the underlying network stack assumes that is always going to be a bytebuf ?